### PR TITLE
Validation errors: Send event on form and support other input elements

### PR DESCRIPTION
**Pull request by kaspernj**
https://app.asana.com/0/1153435032856801/1154591174053995

### CODE REVIEW

File: lib/api_maker/javascript/bootstrap/checkbox.jsx [hunk 1]
```diff
@@ -27,12 +27,32 @@ export default class BootstrapCheckbox extends React.Component {
...
+    this.setForm()
+  }
+
+  componentDidUpdate() {
+    this.setForm()
+  }
+
+  setForm() {
+    const form = this.refs.input && this.refs.input.form
+    if (form != this.state.form) this.setState({form})
```
Comment (AndersGM) [inline lib/api_maker/javascript/bootstrap/checkbox.jsx]:
```suggestion
    if (form !== this.state.form) this.setState({form})
```

File: lib/api_maker/javascript/bootstrap/radio-buttons.jsx [hunk 2]
```diff
@@ -17,11 +19,35 @@ export default class BootstrapRadioButtons extends React.Component {
...
+    this.setForm()
+  }
+
+  componentDidUpdate() {
+    this.setForm()
+  }
+
+  setForm() {
+    const form = this.refs.hiddenInput && this.refs.hiddenInput.form
+    if (form != this.state.form) this.setState({form})
```
Comment (AndersGM) [inline lib/api_maker/javascript/bootstrap/radio-buttons.jsx]:
```suggestion
    if (form !== this.state.form) this.setState({form})
```

File: lib/api_maker/javascript/bootstrap/select.jsx [hunk 3]
```diff
@@ -40,6 +40,8 @@ export default class BootstrapSelect extends React.Component {
   }
 
   componentDidMount() {
+    this.setForm()
+
```
Comment (AndersGM) [inline lib/api_maker/javascript/bootstrap/select.jsx]:
```suggestion
    const { onChange, select, select2 } = this.props
```

File: lib/api_maker/javascript/bootstrap/select.jsx [hunk 4]
```diff
@@ -40,6 +40,8 @@ export default class BootstrapSelect extends React.Component {
...
     if (this.props.select2 && this.props.onChange)
```
Comment (AndersGM) [inline lib/api_maker/javascript/bootstrap/select.jsx]:
```suggestion
    if (select2 && onChange)
```

File: lib/api_maker/javascript/bootstrap/select.jsx [hunk 5]
```diff
@@ -40,6 +40,8 @@ export default class BootstrapSelect extends React.Component {
...
       $(this.refs.select).on("change", this.props.onChange)
```
Comment (AndersGM) [inline lib/api_maker/javascript/bootstrap/select.jsx]:
```suggestion
      $(this.refs.select).on("change", onChange)
```

File: lib/api_maker/javascript/bootstrap/select.jsx [hunk 6]
```diff
@@ -53,14 +55,21 @@ export default class BootstrapSelect extends React.Component {
       $(this.refs.select).off("change", this.props.onChange)
```
Comment (AndersGM) [inline lib/api_maker/javascript/bootstrap/select.jsx]:
```suggestion
      $(this.refs.select).off("change", onChange)
```

File: spec/dummy/app/javascript/api-maker/bootstrap/radio-buttons.jsx [hunk 7]
```diff
@@ -17,11 +19,35 @@ export default class BootstrapRadioButtons extends React.Component {
...
+    this.setForm()
+  }
+
+  componentDidUpdate() {
+    this.setForm()
+  }
+
+  setForm() {
+    const form = this.refs.hiddenInput && this.refs.hiddenInput.form
+    if (form != this.state.form) this.setState({form})
```
Comment (AndersGM) [inline spec/dummy/app/javascript/api-maker/bootstrap/radio-buttons.jsx]:
```suggestion
    if (form !== this.state.form) this.setState({form})
```

File: lib/api_maker/javascript/bootstrap/checkbox.jsx (hunk 1 above)
Comment (kaspernj) [reply lib/api_maker/javascript/bootstrap/checkbox.jsx]: What is the advantage going to be in this case?

File: lib/api_maker/javascript/bootstrap/radio-buttons.jsx (hunk 2 above)
Comment (kaspernj) [reply lib/api_maker/javascript/bootstrap/radio-buttons.jsx]: Same

**Pull request merged**